.github/workflows/build.yml

```diff
@@ -9,24 +9,35 @@ on:
 jobs:
   generate:
     runs-on: ubuntu-latest
-
     strategy:
       matrix:
         node-version: [12.x]
-
     steps:
     - uses: actions/checkout@v1
     - name: Use Node.js ${{ matrix.node-version }}
       uses: actions/setup-node@v1
       with:
         node-version: ${{ matrix.node-version }}
-    - name: npm install, build and lint
-      run: |
-        npm ci
-        npm start
-        npm run lint
+    - run: npm ci
+    - name: Generate types
+      run: npm start
+    - run: npm run lint
       env:
-        CI: true
+        GAPI_MAX_PARALLEL: 3
+    - name: Upload types
+      uses: actions/upload-artifact@v1
+      with:
+        name: types
+        path: types
```

Comment:
Add `.tmp` folder to artifacts, please

Reply:
> My problem with leaving JSON in the types is that we don't want to publish this JSON to the DT.

Yeah, I hear you – we would just have to ensure the JSON is removed or not generated in this build context. With that approach, I was assuming we would only generate the JSON in the linting context. Would we also want to generate it here as well?

Reply:
Not sure what you mean by "linting context"... We can only save JSON during the generation step (when we actually download JSON to generate types). So, to have JSON available during the linting step - we need to save it in the generation step and persist it across both of these steps.

Another thing is that we only have linting step in one action that is ran for both new PRs and by cron. I want to have JSON available for investigation when this action fails (no matter if it's triggered by PR or cron). Basically, if linting ever fails - I want to know exactly why :)

Reply:
> Not sure what you mean by "linting context"

For posterity, I was referring to the `CI Lint Types` workflow, but you're right: `npm run lint` could certainly fail in the `Build` workflow as well.